plugin/trino-opa/src/test/java/io/trino/plugin/opa/TestOpaBatchAccessControlFiltering.java

```diff
@@ -49,226 +52,235 @@
+        );
+
+        for (TestCase testCase : testCases) {
+            InstrumentedHttpClient mockClient = createMockHttpClient(
+                    OPA_SERVER_BATCH_URI,
+                    buildHandler(pathToResources,
+                            ImmutableSet.of("\"table_one_column_1\"", "\"table_one_column_2\"", "\"table_one_column_3\"", "\"table_two_column_1\"")
+                                    .stream()
+                                    .map(RequestTestUtilities::toJsonNode)
+                                    .collect(toImmutableSet())));
+
+            OpaAccessControl authorizer = createOpaAuthorizer(testCase.opaConfig, mockClient);
+            Map<SchemaTableName, Set<String>> result = authorizer.filterColumns(
+                    TEST_SECURITY_CONTEXT,
+                    catalog,
+                    requestedColumns);
+
+            assertThat(result).containsExactlyInAnyOrderEntriesOf(
+                    ImmutableMap.of(tableOne, ImmutableSet.of("table_one_column_1", "table_one_column_2", "table_one_column_3"),
+                            tableTwo, ImmutableSet.of("table_two_column_1")));
+
+            assertRequestMatchesExpectedRequestProperties(new ExpectedRequestProperties(
+                            testCase.batchSizeToBatchNumber,
+                            testCase.numberOfRequest,
+                            "FilterColumns",
+                            requestedColumns
+                                    .entrySet()
+                                    .stream()
+                                    .flatMap(entry -> entry.getValue().stream())
+                                    .map(item -> toJsonNode("\"%s\"".formatted(item)))
+                                    .collect(toImmutableSet())),
+                    ImmutableList.copyOf(mockClient.getRequests()),
+                    pathToResources);
+        }
     }
 
     @Test
     public void testEmptyFilterColumns()
     {
+        String catalog = "my_catalog";
+
         assertFilteringAccessControlMethodDoesNotSendRequests(
-                accessControl -> accessControl.filterColumns(TEST_SECURITY_CONTEXT, "my_catalog", ImmutableMap.of()).entrySet());
+                accessControl -> accessControl.filterColumns(TEST_SECURITY_CONTEXT, catalog, ImmutableMap.of()).entrySet());
 
         SchemaTableName tableOne = SchemaTableName.schemaTableName("my_schema", "table_one");
         SchemaTableName tableTwo = SchemaTableName.schemaTableName("my_schema", "table_two");
-        Map<SchemaTableName, Set<String>> requestedColumns = ImmutableMap.<SchemaTableName, Set<String>>builder()
-                .put(tableOne, ImmutableSet.of())
-                .put(tableTwo, ImmutableSet.of())
-                .buildOrThrow();
+        Map<SchemaTableName, Set<String>> requestedColumns = ImmutableMap.of(
+                tableOne, ImmutableSet.of()
+                , tableTwo, ImmutableSet.of());
```

Comment:
Formatting here is a bit odd
```suggestion
        Map<SchemaTableName, Set<String>> requestedColumns = ImmutableMap.of(
                tableOne, ImmutableSet.of(),
                tableTwo, ImmutableSet.of());
```